src/server/analytic_services/prometheus_reporting.js

```diff
@@ -461,25 +514,33 @@ class PrometheusReporting {
     }
 
     set_resource_status(resources_info) {
-        if (!this.enabled()) return;
         this._metrics.resource_healthy.reset();
         resources_info.forEach(resource_info => {
             this._metrics.resource_healthy.set({ resource_name: resource_info.resource_name }, Number(resource_info.is_healthy));
         });
     }
 
     update_providers_bandwidth(type, write_size, read_size) {
-        if (!this.enabled()) return;
         this._metrics.providers_bandwidth_read_size.inc({ type }, read_size);
         this._metrics.providers_bandwidth_write_size.inc({ type }, write_size);
     }
 
     update_providers_ops(type, write_num, read_num) {
-        if (!this.enabled()) return;
         this._metrics.providers_ops_read_num.inc({ type }, read_num);
         this._metrics.providers_ops_write_num.inc({ type }, write_num);
     }
 }
 
-// EXPORTS
-exports.PrometheusReporting = PrometheusReporting;
+register_report(
```

Comment:
What is the purpose of prometheus_reporting.js? It seems that we should refactor this file. If endpoint imported it, endpoint would register NooBaaCore metrics. We would not want that.  We can,
- Only keep the common functions (e.g. start_server) and class BasePrometheusReport in the file
- Move inheriting class (e.g.) NooBaaCoreReport to the space of its corresponding service
- Move calling register_report() to the space of corresponding service

Reply:
prometheus_reporting.js is the overall service which gives you the ability access, export, and exposes reports over the network for scraping and endpoint should import it.

Registring a report does not mean that we are collecting its metrics or even instantiating the report class.
If you look at:
```
function get_report(name, create_non_existing = true) {
    const report = reports.get(name);
    if (!report) {
        throw new Error(`get_report() - No report registered for name ${name}`);
    }

    if (!report.instance && create_non_existing) {
        report.instance = new report.Type();
    }
    return report.instance;
}
```

You can see that a report is instantiated only if you try to access it, so which report/metrics are collected and exposed depends on the code that runs in a give process.

Reply:
@paulxuzhang 
I just like to add that I pushed a commit that split prometheus_reporting.js and moves reports codes into their own files.
This is not exactly what you suggested as we are still registering all reports but I think my last comment can explain the reasoning